backend/tests/v5/e2e/routes/users.test.js

```diff
@@ -152,16 +156,225 @@ const testGetUsername = () => {
 	});
 };
 
+const formatUserProfile = (user, hasAvatar = true) => ({
+	username: user.user,
+	firstName: user.basicData.firstName,
+	lastName: user.basicData.lastName,
+	email: userEmail,
+	apiKey: user.apiKey,
+	hasAvatar,
+});
+
+const testGetProfile = () => {
+	describe('Get profile of the logged in user', () => {
+		test('should fail if the user is not logged in', async () => {
+			const res = await agent.get('/v5/user/').expect(templates.notLoggedIn.status);
+			expect(res.body.code).toEqual(templates.notLoggedIn.code);
+		});
+
+		test('should return the user profile if the user is logged in', async () => {
+			await testSession.post('/v5/login/').send({ user: testUser.user, password: testUser.password });
+			const res = await testSession.get('/v5/user/').expect(200);
+			expect(res.body).toEqual(formatUserProfile(testUser));
+		});
+
+		test('should return the user profile if the user has a session via an API key', async () => {
+			const res = await agent.get(`/v5/user?key=${testUser.apiKey}`).expect(200);
+			expect(res.body).toEqual(formatUserProfile(testUser));
+		});
+	});
+};
+
+const testUpdateProfile = () => {
+	describe('Update profile of the logged in user', () => {
+		test('should fail if the user is not logged in', async () => {
+			const data = { firstName: 'newName' };
+			const res = await agent.put('/v5/user/').send(data).expect(templates.notLoggedIn.status);
+			expect(res.body.code).toEqual(templates.notLoggedIn.code);
+		});
+
+		test('should fail if the user has a session via an API key', async () => {
+			const data = { firstName: 'newName' };
+			const res = await agent.put(`/v5/user?key=${testUser.apiKey}`)
+				.send(data).expect(templates.notLoggedIn.status);
+			expect(res.body.code).toEqual(templates.notLoggedIn.code);
+		});
+
+		test('should fail if the update data have invalid email', async () => {
```

Comment:
might be better to group these ones (the ones where you have logged in) into another describe, so you can [do some setup and tear down ](https://jestjs.io/docs/setup-teardown), something like 
````js
describe("With valid authentication", () => {
    beforeAll(async () => {
        //log in before doing all the tests that assumes the user has logged in
        await testSession.post('/v5/login/').send({ user: testUser.user, password: testUser.password });
   });
    afterAll(async() => {
         // log out after this block of tests to ensure we don't contaminate other test cases
        await testSession.post('/v5/logout/');
    });

    test(/*...*/);
    test(/*...*/);
    test(/*...*/);
})
````

Describe can be nested so you should be able to just have another describe block here